spharpy/classes/audio.py

```diff
@@ -0,0 +1,178 @@
+        sided spectra, i.e., for all frequencies between 0 Hz and
+        half the sampling rate.
+    sampling_rate : double
+        Sampling rate in Hz
+    basis_type : str
+        Type of spherical harmonic basis, either ``'complex'`` or
+        ``'real'``.
+    normalization : str
+        Normalization convention, either ``'n3d'``, ``'maxN'`` or
+        ``'sn3d'``. (maxN is only supported up to 3rd order)
+    channel_convention : str
+        Channel ordering convention, either ``'acn'`` or ``'fuma'``.
+        (FuMa is only supported up to 3rd order)
+    condon_shortley : bool or str, optional
+        Whether to include the Condon-Shortley phase term. If ``True``
+        Condon-Shortley is included, if ``False`` it is not included.
+    n_samples : int, optional
+        Number of time domain samples. Required if domain is ``'freq'``.
+        The default is ``None``, which assumes an even number of samples
+        if the data is provided in the frequency domain.
+    domain : ``'time'``, ``'freq'``, optional
+        Domain of data. The default is ``'time'``
+    fft_norm : str, optional
+        The normalization of the Discrete Fourier Transform (DFT). Can be
+        ``'none'``, ``'unitary'``, ``'amplitude'``, ``'rms'``, ``'power'``,
+        or ``'psd'``. See :py:func:`~pyfar.dsp.fft.normalization` and [#]_
+        for more information. The default is ``'none'``, which is typically
+        used for energy signals, such as impulse responses.
+    comment : str
+        A comment related to `data`. The default is ``None``.
+    is_complex : bool, optional
+        Specifies if the underlying time domain data are complex
+        or real-valued. If ``True`` and `domain` is ``'time'``, the
+        input data will be cast to complex. The default is ``False``.
+
+    References
+    ----------
+    .. [#] F. Zotter, M. Frank, "Ambisonics A Practical 3D Audio Theory
+            for Recording, Studio Production, Sound Reinforcement, and
+            Virtual Reality", (2019), Springer-Verlag
+    .. [#] B. Rafely, "Fundamentals of Spherical Array Processing", (2015),
+            Springer-Verlag
+    .. [#] E.G. Williams, "Fourier Acoustics", (1999), Academic Press
+
+    """
+    def __init__(
+            self,
+            data,
+            sampling_rate,
+            basis_type,
+            normalization,
+            channel_convention,
+            condon_shortley,
+            n_samples=None,
+            domain='time',
+            fft_norm='none',
+            comment="",
+            is_complex=False):
+        """
+        Create SphericalHarmonicSignal with data, and sampling rate.
+        """
+        # check dimensions
+        if len(data.shape) < 3:
+            raise ValueError("Invalid number of dimensions. Data should have "
+                             "at least 3 dimensions.")
+
+        # set n_max
+        self._n_max = int(np.sqrt(data.shape[-2]))-1
```

Comment:
We should add a check, to make sure the number of channels is valid, e.g.
```suggestion
        n_max = np.sqrt(data.shape[-2])-1
        if n_max - int(n_max) != 0:
            raise ValueError('Bla')
        self._n_max = n_max
```